### PR TITLE
Do not apply minimum-should-match on a boolean query if the coords are disabled

### LINKED ISSUE #15858: Minimum_should_match for one word query
The following is a search query. The term full_name uses a synonym analyzer. 
The synonym expand file has the synonym: rice,arroz. 
Now if we run the validate query we get the below results:

    GET /products_v2/product/_validate/query?explain
    {
      "query": {
        "bool": {
          "must": [
            {
              "match": {
                "full_name": {
                  "query": "rice",
                  "fuzziness": 1,
                  "minimum_should_match": "3<90%"
                }
              }
            }
          ],
          "should": [
            {
              "match_phrase": {
                "name": {
                  "query": "rice",
                  "slop": 50
                }
              }
            }
          ]
        }
      }
    }

    {
       "valid": true,
       "_shards": {
          "total": 1,
          "successful": 1,
          "failed": 0
       },
       "explanations": [
          {
             "index": "products_v2",
             "valid": true,
             "explanation": "filtered(+((full_name:arroz~1 full_name:rice~1)~2) (name:arroz name:rice))->cache(_type:product)"
          }
       ]
    }

Both rice and arroz are position 1. The minimum_should_match doesn't get applied to the position but the number of words in position 1.

Let's say the synonym file has another synonym :  Brown Sugar, brownsugar.
The following is the result of the validate query calling brownsugar

    {
       "valid": true,
       "_shards": {
          "total": 1,
          "successful": 1,
          "failed": 0
       },
       "explanations": [
          {
             "index": "products_v4",
             "valid": true,
             "explanation": "filtered(+(((full_name:brownsugar~1 full_name:brown~1) full_name:sugar~1)~2) name:\"(brownsugar brown) sugar\"~50)->cache(_type:product)"
          }
       ]
    }

In this case the minimum to match is applied based on positions. brownsugar and brown are position 1, sugar is position 2. The minimum_should_match gets applied as 2.

If minimum_should_match gets applied based on the number of positions there are, I believe the above example for rice should have minimum_should_match should be 1 and not 2. This is a problem in the way ES handles minimum_should_match for one word synonyms.

### PR DESCRIPTION
**Pull request by jimczi**
Fixes #15858

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java [hunk 1]
```diff
@@ -117,6 +117,9 @@ public static Query applyMinimumShouldMatch(BooleanQuery query, @Nullable String
         if (minimumShouldMatch == null) {
             return query;
         }
+        if (query.isCoordDisabled()) {
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java]: can you leave a comment to explain that this means that this booleanquery was generated from synonyms?

**Comment by jpountz** [comment]
@jimferenczi I suspect that this does not fully address #15858. For instance if you pass a min should match of 2 on a match query that only emits a asingle term then the min should match will simply be ignored?

**Comment by jimferenczi** [comment]
@jpountz yes but not because of this commit, a match query that emits a single term is translated into a TermQuery and not a BooleanQuery. The match query builder ignores TermQuery (and any Query that is not a BooleanQuery after parsing) when the minimum should match is applied.

**Comment by jpountz** [comment]
OK, so let's merge this in and work on another PR to make the min should match apply in all cases?

**Pull request merged**